src/frontend/js/modules/datetime/date-picker/date-picker.controller.js

```diff
@@ -33,8 +33,18 @@ require('../datetime.service.js');
       $scope.$on('$destroy', function() {
         unsubscribeFromWatchNgModel();
       });
-
       self.uiValue = self.ngModel && esnDatetimeService.updateObjectToBrowserTimeZone(self.ngModel);
+
+      // Datepicker translation
+      const saveDate = angular.copy(self.uiValue);
+
+      $timeout(function() {
+        self.someChange = true;
+        $timeout(function() {
+          self.someChange = false;
+          self.uiValue = saveDate;
+        }, 150);
+      }, 150);
```

Comment:
Is there a reason why you need `$timeout` here? And why have you chosen `150`?

Reply:
It is not the best way to do the translation , but the problem is with angular-strap  datepicker used in esn  doesn't watch $locale changes , $timeout is used to force the rerender of our component with the new locale,

Reply:
Why did you pick `150` though?

Reply:
and about the 150 it is just for the timeout we can choose any number , it is not important

Reply:
Can we choose `0`, a.k.a leaving the second parameter out?

Reply:
yes we can choose 0 , will update it